libraries/AP_GPS/AP_GPS.h

```diff
@@ -354,6 +354,15 @@ class AP_GPS
         return ground_speed(primary_instance);
     }
 
+    // 3D speed in m/s
+    float speed_3d(uint8_t instance) const WARN_IF_UNUSED {
+        const auto vel = velocity(instance);
+        return norm(vel[0], vel[1], vel[2]);
+    }
+    float speed_3d() const WARN_IF_UNUSED {
+        return speed_3d(primary_instance);
```

Comment:
Why not just use `velocity().length()` ? You can also get 2D with: `velocity().xy().length()`

Reply:
Used velocity().length in 64164f42cf4b37e4a27e5ed38c2b9a2deb0b2eb3.

For 2D, I'm hesitant to change the calculation, because it's already populated by the internal GPS driver.

For example, GSOF does this:
`state.ground_speed = SwapFloat(gsof_msg.data, a + 1);`
ERB does this:
`state.ground_speed     = _buffer.vel.speed_2d * 0.01f;        // m/s`
MAV does this:
`state.ground_speed = packet.vel * 0.01f;`

And more. 

If we use what arrives from the state() call, it's already computed. Trying to avoid any extra sqrt() calls.

Reply:
Sorry, I meant in the airspeed code, then we don't need the new methods. Unless you can find some other places that could use the new methods.